### ISSUE TITLE
MA0182 false positive using delegate

### ISSUE THREAD

**Issue by dmitriy-maksimov-wd** [meziantou/Meziantou.Analyzer#967]
### Version of the Meziantou.Analyzer NuGet package

2.0.277

### Rule Identifier

MA0182

### Target Framework

Net 10.0

### C# Language version

C# 12

### Description

Reported error

```
Error MA0182 : Internal type 'tryAcquireLockDelegate' is apparently never used. If so, remove it from the assembly. If this type is intended to contain only static members, make it 'static'. (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0182.md)
```

but `tryAcquireLockDelegate` in use in Mock `.Callback`

### Reproduction Steps

Minimal code:

```c#
[TestFixture]
[TestOf(typeof(Program))]
public class ProgramTest
{
	public interface IInterface
	{
		bool TryAcquireLock(string key, string comment, TimeSpan timeout, out string lockToken);
	}

	private delegate void tryAcquireLockDelegate(string key, string comment, TimeSpan timeout, out string lockToken);

	[Test]
	public void Test()
	{
		var mockLockManagerClient = new Mock<IInterface>();

		string outParam;
		mockLockManagerClient
			.Setup(x => x.TryAcquireLock(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<TimeSpan>(), out outParam))
			.Returns(true)
			.Callback(new tryAcquireLockDelegate((string key, string comment, TimeSpan timeout, out string lockToken) =>
			{
				lockToken = "42";
			}));
	}
}
```
